components/builder-originsrv/src/migrations/origin_packages.rs

```diff
@@ -267,5 +267,35 @@ pub fn migrate(migrator: &mut Migrator) -> Result<()> {
                         RETURN;
                     END
                     $$ LANGUAGE plpgsql STABLE"#)?;
+    migrator.migrate("originsrv",
+              r#"CREATE OR REPLACE FUNCTION insert_origin_package_v2 (
+                 op_origin_id bigint,
+                 op_owner_id bigint,
+                 op_name text,
+                 op_ident text,
+                 op_checksum text,
+                 op_manifest text,
+                 op_config text,
+                 op_target text,
+                 op_channel text,
+                 op_deps text,
+                 op_tdeps text,
+                 op_exposes text
+              ) RETURNS SETOF origin_packages AS $$
+                  DECLARE
+                     inserted_package origin_packages;
+                     channel_id bigint;
+                  BEGIN
+                      INSERT INTO origin_packages (origin_id, owner_id, name, ident, checksum, manifest, config, target, deps, tdeps, exposes)
+                             VALUES (op_origin_id, op_owner_id, op_name, op_ident, op_checksum, op_manifest, op_config, op_target, op_deps, op_tdeps, op_exposes)
+                             RETURNING * into inserted_package;
+
+                      SELECT id FROM origin_channels WHERE origin_id = op_origin_id AND name = op_channel INTO channel_id;
+                      PERFORM promote_origin_package_v1(channel_id, inserted_package.id);
```

Comment:
I think it would be better to perform these 2 operations - insert and promote - at the api layer rather in the proc. It makes the promote clearer to spot and removes logic from the db.